p2p/security/tls/transport_test.go

```diff
@@ -175,6 +176,83 @@ func TestHandshakeSucceeds(t *testing.T) {
 	})
 }
 
+type testcase struct {
+	clientProtos   []protocol.ID
+	serverProtos   []protocol.ID
+	expectedResult string
+}
+
+func TestHandshakeWithNextProtoSucceeds(t *testing.T) {
+
+	tests := []testcase{
+		{clientProtos: nil, serverProtos: nil, expectedResult: ""},
+		{[]protocol.ID{"muxer1/1.0.0", "muxer2/1.0.1"}, []protocol.ID{"muxer2/1.0.1", "muxer1/1.0.0"}, "muxer2/1.0.1"},
+		{[]protocol.ID{"muxer1/1.0.0", "muxer2/1.0.1", "libp2p"}, []protocol.ID{"muxer2/1.0.1", "muxer1/1.0.0", "libp2p"}, "muxer2/1.0.1"},
+		{[]protocol.ID{"muxer1/1.0.0", "libp2p"}, []protocol.ID{"libp2p"}, ""},
+		{[]protocol.ID{"libp2p"}, []protocol.ID{"libp2p"}, ""},
+		{[]protocol.ID{"muxer1"}, []protocol.ID{}, ""},
+		{[]protocol.ID{}, []protocol.ID{"muxer1"}, ""},
+		{[]protocol.ID{"muxer2"}, []protocol.ID{"muxer1"}, ""},
+	}
+	var expectedMuxer string
```

Comment:
Nit: don't declare this as a (test-) global variable, make it a parameter of `handshake` instead.

Reply:
@marten-seemann thanks for reviewing it again. Good suggestion. updated the variable. Please take a look!